From e947195e4aca517344e7cd77cc4215531f85f3c3 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Wed, 15 Nov 2017 13:40:26 +0200
Subject: [PATCH 189/448] staging: fsl-dpaa2/eth: Only linear skbs require Tx headroom

For non-linear skbs we build SG frames, so there's no requirement
for skb headroom size.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
index 2e2d10c..91e1d5a 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
@@ -822,7 +822,9 @@ static netdev_tx_t dpaa2_eth_tx(struct sk_buff *skb, struct net_device *net_dev)
 	percpu_stats = this_cpu_ptr(priv->percpu_stats);
 	percpu_extras = this_cpu_ptr(priv->percpu_extras);
 
-	if (skb_headroom(skb) < dpaa2_eth_tx_headroom(priv)) {
+	/* For non-linear skb we don't need a minimum headroom */
+	if (skb_headroom(skb) < dpaa2_eth_tx_headroom(priv) &&
+	    !skb_is_nonlinear(skb)) {
 		struct sk_buff *ns;
 
 		ns = skb_realloc_headroom(skb, dpaa2_eth_tx_headroom(priv));
-- 
1.7.1

